src/components/LanguageSwitch/LanguageSwitch.tsx

```diff
@@ -42,3 +42,26 @@ export const LanguageSwitch: React.FC<LanguageSwitchProps> = props => {
 };
 
 LanguageSwitch.displayName = "LanguageSwitch";
+
+export const LanguageSwitchWithCaching = (props: LanguageSwitchProps) => {
+  const { cachedValues, pushValue } = useCachedLocales();
+
+  const cachedWithLabel = cachedValues
+    .map(code => props.languages.find(l => l.code === code))
+    .filter(v => !!v) as LanguageFragment[];
```

Comment:
tip: We could use this to filter out missing values this should remove need for type casting.

```suggestion
    .filter(Boolean);
```